main.go

```diff
@@ -45,39 +42,54 @@ func main() {
 	}
 	cs := kubernetes.NewForConfigOrDie(config)
 
-	watch, err := cs.CoreV1().Pods("").Watch(context.Background(), metav1.ListOptions{})
+	watch, err := cs.CoreV1().Pods("").Watch(ctx, metav1.ListOptions{})
 
 watchloop:
 	for evt := range watch.ResultChan() {
 		pod := evt.Object.(*v1.Pod)
-		switch evt.Type {
-		case "ADDED", "MODIFIED":
-			for _, status := range pod.Status.ContainerStatuses {
-				if w := status.State.Waiting; w != nil {
-					if w.Reason == "CrashLoopBackOff" {
-						log.Printf("Pod %v/%v is in CrashLoopBackOff, deleting", pod.Namespace, pod.Name)
-						for {
-							if _, _, err = chat.PostMessage(*channelId, slack.MsgOptionText(
-								fmt.Sprintf("Deleting crashlooping pod `%s/%s`", pod.Namespace, pod.Name), false)); err != nil {
-								var rlError *slack.RateLimitedError
-								if errors.As(err, &rlError) {
-									log.Printf("Rate limited, sleeping %s", rlError.RetryAfter)
-									time.Sleep(rlError.RetryAfter)
-									break
+		if v, found := pod.Labels["crashloopbackon"]; found {
+			if b, _ := strconv.ParseBool(v); !b {
+				continue watchloop
+			}
+			switch evt.Type {
+			case "ADDED", "MODIFIED":
+				if v := pod.Labels["crashloopbackon"]; v == "true" {
+					for _, status := range pod.Status.ContainerStatuses {
+						if w := status.State.Waiting; w != nil {
+							if w.Reason == "CrashLoopBackOff" {
+								if err := handle(pod, cs, ctx, ddEvents, status.Name); err != nil {
+									log.Fatalf("Could not handle %v: %v", pod, err)
 								}
-								log.Fatalf("Could not post Slack message: %v", err)
-							} else {
-								break
+								continue watchloop
 							}
 						}
-						err := cs.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
-						if err != nil {
-							log.Printf("Could not delete %v/%v: %v", pod.Namespace, pod.Name, err)
-						}
-						continue watchloop
 					}
 				}
 			}
 		}
 	}
 }
+
+func handle(pod *v1.Pod, cs *kubernetes.Clientset, ctx context.Context, events *datadogV1.EventsApi, containerName string) error {
+	tags := []string{
+		fmt.Sprintf("namespace:%s", pod.Namespace),
+		fmt.Sprintf("pod_name:%s", pod.Name),
+		fmt.Sprintf("workload:%s", containerName),
+	}
+	if pteam, found := pod.Labels["com.apollographql/primaryTeam"]; found {
+		tags = append(tags, fmt.Sprintf("team:%s", pteam))
+	}
+	if _, _, err := events.CreateEvent(ctx, datadogV1.EventCreateRequest{
```

Comment:
Not sure I remember the implication of DD events or how they'll show up (or why better than Slack directly)

Reply:
Dropping the move to DD.